### PR TITLE
RoR2: Add Support for New Stage

### PR DESCRIPTION
**Pull request by kindasneaki**
## What is this fixing or adding?
Adds support for the newly added stage in Risk of Rain 2. This stage will be available to both vanilla and the DLC.
Also this adds in option groups to better visualize the options for the two different modes that are select-able.

## How was this tested?
Played a solo world and will be releasing the apworld, so other players will be able to test/play the new stage in AP.

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

**Review by Exempt-Medic — APPROVED** [review]
Merged into `main`, got the new client, tested that the new stage worked (it could be gone to, left from, it worked as a received item and as a starting stage, and checks within it could be sent out)

**Review by nicholassaylor — CHANGES_REQUESTED** [review]
Option groups did not appear on Webhost because the changes aren't on this branch.

**Comment by kindasneaki** [comment]
> Option groups did not appear on Webhost because the changes aren't on this branch.

Guess I forgot to hit push... Thanks for catching that.

File: worlds/ror2/options.py [hunk 1]
```diff
@@ -375,6 +375,44 @@ class ItemWeights(Choice):
...
+    OptionGroup("Classic Mode Options", [
+        TotalLocations,
+    ], start_collapsed=True),
+    OptionGroup("Weighted Choices", [
+        ItemWeights,
+        ItemPoolPresetToggle,
+        GreenScrap,
+        RedScrap,
+        YellowScrap,
+        WhiteScrap,
```
Comment (nicholassaylor) [inline worlds/ror2/options.py:397]:
```suggestion
        WhiteScrap,
        GreenScrap,
        RedScrap,
        YellowScrap,
```
To match the order of the items below

**Review by nicholassaylor — APPROVED** [review]
One tiny nitpick, but looks good. Tested generation and Webhost changes but not client.

File: worlds/ror2/options.py (hunk 1 above)
Comment (kindasneaki) [reply worlds/ror2/options.py]: Thanks. Fun fact (just learned), the order doesn't matter in the group. It goes off of the dataclass order

**Review by NewSoupVi — APPROVED** [review]

**Pull request merged**